plugin/mlsdbprovider.cpp

```diff
@@ -97,7 +103,11 @@ MlsdbProvider::MlsdbProvider(QObject *parent)
     connect(m_cellWatcher, &QOfonoExtCellWatcher::cellsChanged,
             this, &MlsdbProvider::cellularNetworkRegistrationChanged);
 
-    cellularNetworkRegistrationChanged();
+    if (m_positioningEnabled) {
+        cellularNetworkRegistrationChanged();
```

Comment:
Not wrong, but alternatively can call calculatePositionAndEmitLocation() directly.